spec/src/modules/tracker.js

```diff
@@ -742,6 +742,102 @@ describe(`ConstructorIO - Tracker${bundledDescriptionSuffix}`, () => {
       expect(tracker.trackAutocompleteSelectV2(term, { ...requiredParameters, ...optionalParameters, ...v2Parameters })).to.equal(true);
     });
 
+    it('Backwards Compatibility - V2 Should respond with a valid response when snake cased parameters are provided', (done) => {
+      const { tracker } = new ConstructorIO({
+        apiKey: testApiKey,
+        fetch: fetchSpy,
+        ...requestQueueOptions,
+      });
+      const snakeCaseParameters = {
+        original_query: 'original-query',
+        section: 'Search Suggestions',
+        tr: 'click',
+        group_id: 'All',
+        display_name: 'display-name',
+        variation_id: '12345-A',
+        item_id: '12345',
+      };
+
+      tracker.on('success', (responseParams) => {
+        const requestParams = helpers.extractUrlParamsFromFetch(fetchSpy);
+        const bodyParams = helpers.extractBodyParamsFromFetch(fetchSpy);
+
+        // Request
+        expect(fetchSpy).to.have.been.called;
+        expect(requestParams).to.have.property('key');
+        expect(requestParams).to.have.property('i');
+        expect(requestParams).to.have.property('s');
+        expect(requestParams).to.have.property('c').to.equal(clientVersion);
+        expect(requestParams).to.have.property('_dt');
+        expect(requestParams).to.have.property('section').to.equal(requiredParameters.section);
+        expect(requestParams).to.have.property('origin_referrer').to.equal('localhost.test/path/name');
+
+        // Body
+        expect(bodyParams).to.have.property('user_input').to.equal(snakeCaseParameters.original_query);
+        expect(bodyParams).to.have.property('tr').to.equal(optionalParameters.tr);
+        expect(bodyParams).to.have.property('item_id').to.equal(snakeCaseParameters.item_id);
+        expect(bodyParams).to.have.property('variation_id').to.equal(snakeCaseParameters.variation_id);
+        expect(bodyParams).to.have.property('section').to.deep.equal(snakeCaseParameters.section);
+        expect(bodyParams).to.have.property('item_name').to.equal(term);
+        expect(bodyParams).to.have.property('group_id').to.equal(optionalParameters.groupId);
+
+        // Response
+        expect(responseParams).to.have.property('method').to.equal('POST');
+        expect(responseParams).to.have.property('message').to.equal('ok');
+
+        done();
+      });
+
+      tracker.on('error', (error) => {
+        done(error);
+      });
+
+      // eslint-disable-next-line max-len
+      expect(tracker.trackAutocompleteSelectV2(term, snakeCaseParameters)).to.equal(true);
+    });
+
+    it('Backwards Compatibility - Should respond with a valid response when snake cased parameters are provided', (done) => {
```

Comment:
I noticed that this backwards compatibility test and the one above it says `...when snake cased parameters are provided`, but the other backwards compatibility tests below says `...when required parameters are provided`.

Should those also say `...when snake cased parameters are provided`?

Reply:
Good catch! I copied the initial one from Node SDK but only added `Backwards Compatibility` to the rest. Should be fixed in [0d68ed7](https://github.com/Constructor-io/constructorio-client-javascript/pull/262/commits/0d68ed7d7e04d8a7fa3e80ee372fa3766278abca)